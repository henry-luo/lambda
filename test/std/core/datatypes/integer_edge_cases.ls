// Test: Integer Edge Cases
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test integer limits and edge cases
let max_int = 9223372036854775807  // 2^63 - 1
let min_int = -9223372036854775808 // -2^63

// Test operations at boundaries
let overflow = max_int + 1  // Should wrap around to min_int
let underflow = min_int - 1 // Should wrap around to max_int

// Test with zero
let zero = 0
let div_by_zero = 1 / 0  // Should be positive infinity
let mod_zero = 1 % 0     // Should be NaN

// Test with one
let one = 1
let power_zero = one ** 0  // Should be 1
let power_one = one ** 1   // Should be 1

// Final check
1
