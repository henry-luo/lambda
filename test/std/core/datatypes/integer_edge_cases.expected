09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/integer_edge_cases.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 21, Col 23 - 21, Col 24: ERROR
09:15:42 [ERROR] ERROR node at Ln 21, Col 23 - 21, Col 24
09:15:42 [ERROR] Syntax error at Ln 22, Col 22 - 22, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 22 - 22, Col 23
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/integer_edge_cases.ls' executed: #####
null
